internacao/src/main/java/br/com/basis/madre/service/LeitoService.java

```diff
@@ -78,14 +83,28 @@ public LeitoDTO save(LeitoDTO leitoDTO) {
     @Transactional(readOnly = true)
     public List<LeitoProjection> getLeitosDesocupadosPor(String nome,
         Pageable pageable) {
-        log.debug("Request to get all Leitos");
         Sort sort = pageable.getSort();
         SituacaoDeLeitoDTO situacao = situacaoDeLeitoService.findByNomeIgnoreCase("Desocupado");
         return leitoRepository
             .findBySituacaoAndNomeIgnoreCaseContaining(situacaoDeLeitoMapper.toEntity(situacao),
                 nome, sort);
     }
 
+    @Transactional(readOnly = true)
+    public List<LeitoProjection> getLeitosNaoDesocupadosPor(String nome,
+        Pageable pageable) {
+        Sort sort = pageable.getSort();
+        SituacaoDeLeitoDTO bloqueado = situacaoDeLeitoService.findByNomeIgnoreCase("bloqueado");
+        SituacaoDeLeitoDTO reservado = situacaoDeLeitoService.findByNomeIgnoreCase("reservado");
+
+        List<SituacaoDeLeito> situacoes = Arrays.asList(bloqueado, reservado).stream()
+            .map(situacaoDeLeitoMapper::toEntity).collect(Collectors.toList());
+
+        return leitoRepository
+            .findBySituacaoInAndNomeIgnoreCaseContaining(situacoes,
```

Comment:
A pesquisa não deveria do tipo like?

Reply:
Eu coloquei IN porque esse método precisa retornar leitos que tem a situação "BLOQUEADO" ou "RESERVADO". Está incorreto?